tests/core/test_model.py

```diff
@@ -965,3 +967,39 @@ class Bar(pa.SchemaModel):
 
     assert Foo.Config.name == "foo"
     assert Bar.Config.name == "Bar"
+
+
+def test_validate_coerce_on_init():
+    """Test that DataFrame[Schema] validates and coerces on initialization."""
+
+    class Schema(pa.SchemaModel):
+        state: Series[str]
+        city: Series[str]
+        price: Series[float] = pa.Field(
+            in_range={"min_value": 5, "max_value": 20}
+        )
+
+        class Config:
+            coerce = True
+
+    class SchemaNoCoerce(Schema):
+        class Config:
+            coerce = False
+
+    raw_data = {
+        "state": ["NY", "FL", "GA", "CA"],
+        "city": ["New York", "Miami", "Atlanta", "San Francisco"],
+        "price": [8, 12, 10, 16],
+    }
+    pandera_validated_df = DataFrame[Schema](raw_data)
+    pandas_df = pd.DataFrame(raw_data)
+    assert pandera_validated_df.equals(Schema.validate(pandas_df))
+    assert isinstance(pandera_validated_df, DataFrame)
+    assert isinstance(pandas_df, pd.DataFrame)
+
+    Schema.Config.coerce = False
```

Comment:
```suggestion
```
This doesn't look like it is needed since you use `SchemaNoCoerce` below? (I haven't actually checked out the branch to check though)

This doesn't test the dtypes explicitly which was a bit confusing at first, but I guess you're really validating the schemamodel does the same thing as `Schema.validate(pandas_df)` which does test the coercion.